src/app/core/header/header.component.ts

```diff
@@ -7,9 +7,21 @@ import { ChangeDetectionStrategy, Component, EventEmitter, Output, Input } from
   styleUrls: ['./header.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class HeaderComponent {
-  @Input() currentFilters: Record<string, unknown[]>;
+export class HeaderComponent implements OnChanges {
+  @Input() filters: Record<string, unknown[]> = { sex: ['Both'], ageRange: [1, 110], BMIRange: [13, 83] };
```

Comment:
I don't think need to repeat the default settings here. Just need a ? guard in the template.

Reply:
Also the type should be unknown, not unknown[].

Reply:
I ran into some problems when it was unknown type. For some reason it won't treat ageRange and BMIRange as arrays because it thinks they're unknown. So I made all the fields arrays just for this part.

Reply:
Hmm, don't like it either way, try `unknown | unknown[]`

Reply:
Didn't work